os/arch/arm/src/imxrt/imxrt_mtd_flash.c

```diff
@@ -0,0 +1,686 @@
+/****************************************************************************
```

Comment:
1. This is mtd driver. 
  1) rename imxrt_mtd_flash.c to hyperflash.c
  2) move this to fs/driver/mtd
2. Put license as a below
    - samsung license
    - nxp license(is your license follow apache 2.0?)
    - nuttx license

Reply:
About imxrt_mtd_flash.c, this is not only for hyperflash but also all the related nor flash used in IMXRT.
I think that the name of hyperflash.c is not proper.

Reply:
Please see comment above.
You have to implement below 
1. mtd driver of hyperflash.
2. spi bus layer
3. spi bus layer's priv layer (flexspi)
4. flexspi maintain layer.

Reply:
You told me that Hyperflash can be added any of spi, that is flex, lpspi.